pydm/widgets/timeplot.py

```diff
@@ -203,16 +203,20 @@ def getCurves(self):
         return [json.dumps(curve.to_dict()) for curve in self._curves]
     
     def setCurves(self, new_list):
-        new_list = [str(i) for i in new_list]
-        #super(PyDMTimePlot, self).clear()
-        self.clearCurves()
-        for curve_dict in new_list:
-            d = json.loads(str(curve_dict))
-            color = d.get('color')
-            if color:
-                color = QColor(color)
-            self.addYChannel(d['channel'], name=d.get('name'), color=color)
-        
+        try:
+            new_list = [str(i) for i in new_list]
+            #super(PyDMTimePlot, self).clear()
+            self.clearCurves()
+            for curve_dict in new_list:
+                d = json.loads(str(curve_dict))
+                color = d.get('color')
+                if color:
+                    color = QColor(color)
+                self.addYChannel(d['channel'], name=d.get('name'), color=color)
+        # if is an invalid curve, ignore
+        except json.decoder.JSONDecodeError:
+            pass
```

Comment:
I think that this need to be changed to not clear the curves and do half of the process if the data is invalid...
Something like this would probably be better:

```python
def setCurves(self, new_list):
    try:
        new_list = [json.loads(str(i)) for I in new_list]
    except son.decoder.JSONDecodeError:
        print("ERROR: Invalid Curve property value.") # Maybe we should raise a ValueError here...
        return
    self.clearCurves()
    for curve_dict in new_list:
         ...
```

Reply:
@hhslepicka This will not solve the problem, because new_list came with a different format (without color, x_channel, y_channel)